File: sukh_1684/problems/minimum_number_of_operations_to_move_all_balls_to_each_box/solution.js
/**
 * @param {string} boxes
 * @return {number[]}
 */
var minOperations = function(boxes) {
    let BoxesArr = boxes.split("").map(element => Number(element));
    let ans = [];
    for(const [i,_] of BoxesArr.entries()){
        let count = 0;
        for(const [j,element] of BoxesArr.entries()){
            if(i!==j && element != 0){
                count = count + Math.abs(i-j);
            }
        }
        ans.push(count);
    }
    return ans;
    
};

File: sukh_1684/problems/minimum_number_of_operations_to_move_all_balls_to_each_box/solution.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:

        balls_right = 0
        balls_sum = 0
        ans = []
        for index,i in enumerate(boxes [1:]):
            index += 1
            if i == '1':
                balls_right += 1
                balls_sum += index
        ans.append(balls_sum)
        
        balls_left = 1 if boxes[0] == '1' else 0
        
        for index, i in enumerate(boxes [1:]):
            index += 1
            ans.append(ans[index-1]  - balls_right + balls_left)
            if i == '1':
                balls_right -= 1
                balls_left += 1
                
        return ans
        
        